### PR TITLE
optimiz concurrency

### PR DESCRIPTION
**Pull request by guol-fnst**
user can customize number of workers via env "GOMAXPROCS"

# What problem are we solving?
optimiz  maxproc for docker env

# How are we solving the problem?
add ENV "GOMAXPROCS" 

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/storage/disk_location.go [hunk 1]
```diff
@@ -208,8 +209,14 @@ func (l *DiskLocation) concurrentLoadingVolumes(needleMapKind NeedleMapKind, con
...
-		workerNum = 10
+	val, ok := os.LookupEnv("GOMAXPROCS")
+	if ok {
+		num, err := strconv.Atoi(val)
+		if err != nil || num < 1 {
+			num = 10
+			glog.Warningf("failed to set worker number from GOMAXPROCS , set to default:10")
+		}
+		workerNum = num
 	}
```
Comment (chrislusf) [inline weed/storage/disk_location.go:224]: If not found, limit to a default 10 with previous logic.

Comment (guol-fnst) [reply weed/storage/disk_location.go:220]: So in this situation,  the worker number is set to 10 if "GOMAXPROCS" not found,  and never set to NumCPU()?

Comment (guol-fnst) [reply weed/storage/disk_location.go:224]: > If not found, limit to a default 10 with previous logic.

So in this situation, the worker number is set to 10 if "GOMAXPROCS" not found, and never set to NumCPU()?

**Comment by chrislusf** [comment]
use previous logic

```
	workerNum := runtime.NumCPU()
	if workerNum <= 10 {
		workerNum = 10
	}
```

**Pull request merged**